vortex-tui/src/browse/ui/layouts.rs

```diff
@@ -237,25 +237,22 @@ fn render_children_list(app: &mut AppState, area: Rect, buf: &mut Buffer) {
             // Use fuzzy matching to rank and filter results
             let matcher = SkimMatcherV2::default();
 
-            // Collect scored matches
-            let mut scored_matches = layout
+            // Collect matches
+            let matches = layout
                 .child_names()
                 .enumerate()
                 .filter_map(|(idx, name)| {
                     matcher
                         .fuzzy_match(&name, &search_filter)
-                        .map(|score| (idx, name.to_string(), score))
+                        .map(|_| (idx, name.to_string()))
                 })
                 .collect_vec();
 
-            // Sort by score (higher is better)
-            scored_matches.sort_by(|a, b| b.2.cmp(&a.2));
```

Comment:
This was the bug. We store a filter state which is a Vec<bool> indicating which children fuzzy-match the filter.

In the Enter key handler, we select the n-th `true` value in the filter. Before for the `GT` filter, this was selecting `NEGATIVE_TRAIN_SITE` because that was the 1158th child, instead of `GT` which was the first result by score, but was the 1181st child in the layout.